hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -333,7 +334,9 @@ private[hudi] object HoodieSparkSqlWriter {
     log.info(s"Registered avro schema : ${schema.toString(true)}")
     val params = parameters.updated(HoodieWriteConfig.AVRO_SCHEMA, schema.toString)
     val writeConfig = DataSourceUtils.createHoodieConfig(schema.toString, path.get, tblName, mapAsJavaMap(params))
-    val hoodieDF = HoodieDatasetBulkInsertHelper.prepareHoodieDatasetForBulkInsert(sqlContext, writeConfig, df, structName, nameSpace)
+    val isGlobalIndex = SparkHoodieIndex.createIndex(writeConfig).isGlobal
```

Comment:
do we need to create the index to really check if its Global? Wondering if there are simpler means. (may be not, given we support even user defined indexes)

Reply:
may be for out of the box index types, we can maintain a static map of IndexType to isGlobal and use it. wdyt?